impress-org/givewp
# Feature: Update visibility conditions in the Field API to use new framework and handle validations

## Description

This PR updates the Field API visibility conditions in a couple ways:
1. It uses the new [Field Conditions](https://github.com/stellarwp/field-conditions) library to define the conditions
2. It adds an `ExcludeUnless` validation rule when conditions are used so that invisible fields aren't validated and included in the final dataset

## Affects

Form Field Manager uses a legacy system which relies on this. I did have to make some changes to that code to continue working. Otherwise, it affects a PR in Next Gen to add visibility conditions, but it isn't implemented yet.

## Testing Instructions

For the legacy FFM stuff:
1. Make a field using FFM
2. Add a visibility condition that's required
3. Open the form and meet the conditions to display the FFM field
4. Manually remove the "required" attribute from the FFM field
5. Leave the field empty and submit the form
6. It _should_ throw an error that the field must be filled

Now, do the same, but leave the "required" attribute and do not meet the visibility conditions. The form should submit successfully since the required field is invisible.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204343603928217
